### DISCUSSION TITLE
Tell us more about IBM Watsonx Assistant Code for Z

### DISCUSSION THREAD

**Discussion by FALLAI-Denis** [IBM/zopeneditor-about#377] [General]
Hi,

Since its version 3.3.0, IBM Z Open Editor offers an interface with [IBM watsonx  Code Assistant for Z](https://www.ibm.com/products/watsonx-code-assistant-z):

_This product provides an end-to-end application developer lifecycle that includes **application discovery and analysis**, **automated code refactoring** and COBOL to Java conversion. **Developers can automatically refactor selected elements of an application and continue modernizing in COBOL**, or selectively transform code to Java leveraging generative AI using a highly tuned state-of-the-art large language model._

In the documents and presentations that I have been able to consult, we mainly present migration and interfacing with Java, which is not our target.

As far as we are concerned, these are the following two points who we would like to explore:
* application discovery and analysis
* code refactoring (COBOL)

Another need we identified concerns the generation of test data by analysis of COBOL source code.

Indeed, we are engaged in a process of refactoring / renewing a legacy COBOL code produced by a code generator (IBM VisualAge Pacbase, same COBOL code as [IBM Rational Programming Patterns](https://www.ibm.com/products/rational-programming-patterns)) to a "native" COBOL code (managed by IBM Z Open Editor and a proprietary aid tool for writing COBOL code based on snippet injection automation and an "intelligent" framework based on copybooks and conditional compilation).

Would the IBM Z Open Editor and IBM Watsonx Code Assistant for Z couple help us in this work?

How to explore / experience this offer?

Thanks.

**Comment by phaumer** [comment]
Hello @FALLAI-Denis. Thanks for your interest in Code Assistant for Z (WCA4Z). Indeed the focus is on Java here and those tests are generated for the Java code, based on the COBOL code. However, the core parts of the "Understand" workflow of WCA4Z are based on IBM Application Discovery and Delivery Intelligence, which is available as a standalone product: <https://www.ibm.com/products/app-discovery-and-delivery-intelligence>.  The key application for the "Refactor" workflow called Refactoring Assistant, however, is only available as part of WCA4Z. It is a REST server and VS Code extension that allows you to graphically display application asset dependencies (from the ADDI database) and systematically extract code into a new programs. Product management tells me that this part of WCA4Z can be purchased at a discount with ADDI if the Transform parts are not needed. I hope this helps.
